Error with Azure CLI 2.46.0 and Bicep if no bicep configuration exists

<!--- 🛑 Please check existing issues first before continuing: https://github.com/Azure/azure-cli/issues --->

### **This is autogenerated. Please review and update as needed.**

## Describe the bug

**Command Name**
`az bicep install`

**Errors:**
```
Installing Bicep CLI v0.14.85...
The command failed with an unexpected error. Here is the traceback:
No section: 'bicep'
Traceback (most recent call last):
  File "D:\a\_work\1\s\build_scripts\windows\artifacts\cli\Lib\site-packages\knack/cli.py", line 233, in invoke
  File "D:\a\_work\1\s\build_scripts\windows\artifacts\cli\Lib\site-packages\azure/cli/core/commands/__init__.py", line 663, in execute
  File "D:\a\_work\1\s\build_scripts\windows\artifacts\cli\Lib\site-packages\azure/cli/core/commands/__init__.py", line 726, in _run_jobs_serially
  File "D:\a\_work\1\s\build_scripts\windows\artifacts\cli\Lib\site-packages\azure/cli/core/commands/__init__.py", line 697, in _run_job
  File "D:\a\_work\1\s\build_scripts\windows\artifacts\cli\Lib\site-packages\azure/cli/core/commands/__init__.py", line 333, in __call__
  File "D:\a\_work\1\s\build_scripts\windows\artifacts\cli\Lib\site-packages\azure/cli/core/commands/command_operation.py", line 121, in handler
  File "D:\a\_work\1\s\build_scripts\windows\artifacts\cli\Lib\site-packages\azure/cli/command_modules/resource/custom.py", line 3648, in install_bicep_cli
  File "D:\a\_work\1\s\build_scripts\windows\artifacts\cli\Lib\site-packages\azure/cli/command_modules/resource/_bicep.py", line 141, in ensure_bicep_installation
  File "D:\a\_work\1\s\build_scripts\windows\artifacts\cli\Lib\site-packages\knack/config.py", line 99, in get
  File "D:\a\_work\1\s\build_scripts\windows\artifacts\cli\Lib\site-packages\knack/config.py", line 94, in get
  File "D:\a\_work\1\s\build_scripts\windows\artifacts\cli\Lib\site-packages\knack/config.py", line 208, in get
  File "configparser.py", line 783, in get
  File "configparser.py", line 1154, in _unify_values
configparser.NoSectionError: No section: 'bicep'
```

## To Reproduce:
Steps to reproduce the behavior. Note that argument values have been redacted, as they may contain sensitive information.

- install Azure CLI 2.46.0
- `az bicep install --version {}`

## Expected Behavior
No error if configuration is not set.

## Environment Summary
```
Windows-10-10.0.17763-SP0
Python 3.10.10
Installer: MSI

azure-cli 2.46.0
```
## Additional Context
This error also happens when running an `az deployment` command passing a bicep file. It seems that the code is not resilient if no configuration for bicep is specified in the config file.

<!--Please don't remove this:-->
<!--auto-generated-->

All deployments failing here as well, on an Agent in a deployment agent pool.

/usr/bin/bash /home/azuredevops/azdoAgent/_work/_temp/azureclitaskscript1678703942369.sh
ERROR: No section: 'bicep'
##[error]Script failed with exit code: 1

Temporary solution:
Until fixed, downgrading version to az cli 2.45 seems to solve the issue. On agent VM run:
`sudo apt-get install azure-cli=2.45.0-1~focal`

Mind: when fixed, upgrade az cli again!

Is this the same issue that I'm facing when using the AzureResourceManagerTemplateDeployment@3 Azure Pipeline task?
My error message for the past week is:
```
##[error]Check out the troubleshooting guide to see if your issue is addressed: https://docs.microsoft.com/en-us/azure/devops/pipelines/tasks/deploy/azure-resource-group-deployment?view=azure-devops#troubleshooting
##[error]Error: Command failed: az bicep build --file /agent/_work/1/s/ARMTemplates/goweb/prod/prod-goweb.bicep
ERROR: The command failed with an unexpected error. Here is the traceback:
ERROR: No section: 'bicep'

```
This is crippling for my ARM deployments! And the workaround by first running this script doesn't do anything for me>
```
az config set bicep.use_binary_from_path=false
az bicep install
```

> Is this the same issue that I'm facing when using the AzureResourceManagerTemplateDeployment@3 Azure Pipeline task? My error message for the past week is:
> 
> ```
> ##[error]Check out the troubleshooting guide to see if your issue is addressed: https://docs.microsoft.com/en-us/azure/devops/pipelines/tasks/deploy/azure-resource-group-deployment?view=azure-devops#troubleshooting
> ##[error]Error: Command failed: az bicep build --file /agent/_work/1/s/ARMTemplates/goweb/prod/prod-goweb.bicep
> ERROR: The command failed with an unexpected error. Here is the traceback:
> ERROR: No section: 'bicep'
> ```
> 
> This is crippling for my ARM deployments! And the workaround by first running this script doesn't do anything for me:
> 
> ```
> az config set bicep.use_binary_from_path=false
> az bicep install
> ```

This workaround works for me with AzureCLI@2 task and setting input: `useGlobalConfig: true`.

--- a/src/azure-cli/azure/cli/command_modules/resource/_bicep.py
+++ b/src/azure-cli/azure/cli/command_modules/resource/_bicep.py
@@ -138,8 +138,8 @@ def ensure_bicep_installation(cli_ctx, release_tag=None, target_platform=None, s
 
         os.chmod(installation_path, os.stat(installation_path).st_mode | stat.S_IXUSR | stat.S_IXGRP | stat.S_IXOTH)
 
-        use_binary_from_path = cli_ctx.config.get("bicep", "use_binary_from_path").lower()
-        if use_binary_from_path in ["0", "no", "false", "off"]:
+        use_binary_from_path = cli_ctx.config.get("bicep", "use_binary_from_path", "if_found_in_ci").lower()
+        if use_binary_from_path not in ["0", "no", "false", "off"]:
             _logger.warning("The configuration value of bicep.use_binary_from_path has been set to 'false'.")
             cli_ctx.config.set_value("bicep", "use_binary_from_path", "false")
 

--- a/src/azure-cli/azure/cli/command_modules/resource/_bicep.py
+++ b/src/azure-cli/azure/cli/command_modules/resource/_bicep.py
@@ -154,7 +154,7 @@ def ensure_bicep_installation(cli_ctx, release_tag=None, target_platform=None, s
         raise ClientRequestError(f"Error while attempting to download Bicep CLI: {err}")
 
 
-def remove_bicep_installation():
+def remove_bicep_installation(cli_ctx):
     system = platform.system()
     installation_path = _get_bicep_installation_path(system)
 
@@ -162,6 +162,11 @@ def remove_bicep_installation():
         os.remove(installation_path)
     if os.path.exists(_bicep_version_check_file_path):
         os.remove(_bicep_version_check_file_path)
+
+    use_binary_from_path = cli_ctx.config.get("bicep", "use_binary_from_path", "if_found_in_ci").lower()
+    if use_binary_from_path in ["0", "no", "false", "off"]:
+        _logger.warning("The configuration value of bicep.use_binary_from_path has been reset")
+        cli_ctx.config.remove_option("bicep", "use_binary_from_path")
 
 
 def is_bicep_file(file_path):

--- a/src/azure-cli/azure/cli/command_modules/resource/custom.py
+++ b/src/azure-cli/azure/cli/command_modules/resource/custom.py
@@ -3649,7 +3649,7 @@ def install_bicep_cli(cmd, version=None, target_platform=None):
 
 
 def uninstall_bicep_cli(cmd):
-    remove_bicep_installation()
+    remove_bicep_installation(cmd.cli_ctx)
 
 
 def upgrade_bicep_cli(cmd, target_platform=None):

--- a/src/azure-cli/azure/cli/command_modules/resource/tests/latest/test_resource_bicep.py
+++ b/src/azure-cli/azure/cli/command_modules/resource/tests/latest/test_resource_bicep.py
@@ -29,6 +29,8 @@ class TestBicep(unittest.TestCase):
 
         with contextlib.suppress(FileNotFoundError):
             remove_bicep_installation(cli_ctx)
+
+        cli_ctx.config.set_value("bicep", "use_binary_from_path", "false")
         with self.assertRaisesRegex(CLIError, 'Bicep CLI not found. Install it now by running "az bicep install".'):
             run_bicep_command(cli_ctx, ["--version"], auto_install=False)
 
@@ -110,6 +112,7 @@ class TestBicep(unittest.TestCase):
         get_bicep_latest_release_tag_stub.return_value = "v2.0.0"
 
         cli_ctx.config.set_value("bicep", "check_version", "True")
+        cli_ctx.config.set_value("bicep", "use_binary_from_path", "false")
         run_bicep_command(cli_ctx, ["--version"])
 
         warning_mock.assert_called_once_with(

--- a/src/azure-cli/azure/cli/command_modules/resource/tests/latest/test_resource_bicep.py
+++ b/src/azure-cli/azure/cli/command_modules/resource/tests/latest/test_resource_bicep.py
@@ -11,6 +11,7 @@ from unittest import mock
 from knack.util import CLIError
 from azure.cli.command_modules.resource._bicep import (
     ensure_bicep_installation,
+    remove_bicep_installation,
     run_bicep_command,
     validate_bicep_target_scope,
     _bicep_version_check_file_path,
@@ -26,6 +27,8 @@ class TestBicep(unittest.TestCase):
     def test_run_bicep_command_raise_error_if_not_installed_and_not_auto_install(self, isfile_stub):
         isfile_stub.return_value = False
 
+        with contextlib.suppress(FileNotFoundError):
+            remove_bicep_installation()
         with self.assertRaisesRegex(CLIError, 'Bicep CLI not found. Install it now by running "az bicep install".'):
             run_bicep_command(cli_ctx, ["--version"], auto_install=False)
 
@@ -87,8 +90,6 @@ class TestBicep(unittest.TestCase):
             "Bicep CLI version 0.13.1 (e3ac80d678)",
         )
 
-        cli_ctx.config.set_value("bicep", "use_binary_from_path", "false")
-
     @mock.patch("azure.cli.command_modules.resource._bicep._logger.warning")
     @mock.patch("azure.cli.command_modules.resource._bicep._run_command")
     @mock.patch("azure.cli.command_modules.resource._bicep.ensure_bicep_installation")
@@ -108,6 +109,7 @@ class TestBicep(unittest.TestCase):
         _get_bicep_installed_version_stub.return_value = "1.0.0"
         get_bicep_latest_release_tag_stub.return_value = "v2.0.0"
 
+        cli_ctx.config.set_value("bicep", "check_version", "True")
         run_bicep_command(cli_ctx, ["--version"])
 
         warning_mock.assert_called_once_with(

--- a/src/azure-cli/azure/cli/command_modules/resource/tests/latest/test_resource_bicep.py
+++ b/src/azure-cli/azure/cli/command_modules/resource/tests/latest/test_resource_bicep.py
@@ -19,7 +19,6 @@ from azure.cli.core.azclierror import InvalidTemplateError
 from azure.cli.core.mock import DummyCli
 
 cli_ctx = DummyCli()
-cli_ctx.config.set_value("bicep", "use_binary_from_path", "false")
 
 
 class TestBicep(unittest.TestCase):
@@ -29,6 +28,35 @@ class TestBicep(unittest.TestCase):
 
         with self.assertRaisesRegex(CLIError, 'Bicep CLI not found. Install it now by running "az bicep install".'):
             run_bicep_command(cli_ctx, ["--version"], auto_install=False)
+
+    @mock.patch("os.chmod")
+    @mock.patch("os.stat")
+    @mock.patch("io.BufferedWriter")
+    @mock.patch("azure.cli.command_modules.resource._bicep.open")
+    @mock.patch("azure.cli.command_modules.resource._bicep.urlopen")
+    @mock.patch("os.path.exists")
+    @mock.patch("os.path.dirname")
+    @mock.patch("os.path.isfile")
+    def test_use_bicep_cli_from_path_false_after_install(self, isfile_stub, dirname_stub, exists_stub, urlopen_stub, open_stub, buffered_writer_stub, stat_stub, chmod_stub):
+        isfile_stub.return_value = False
+        dirname_stub.return_value = "tmp"
+        exists_stub.return_value = True
+        buffered_writer_stub.write.return_value = None
+
+        stat_result = mock.Mock()
+        stat_result.st_mode = 33206
+        stat_stub.return_value = stat_result
+
+        chmod_stub.return_value = None
+
+        response = mock.Mock()
+        response.getcode.return_value = 200
+        response.read.return_value = b"test"
+        urlopen_stub.return_value = response
+
+        ensure_bicep_installation(cli_ctx, release_tag="v0.14.85", stdout=False)
+
+        self.assertTrue(cli_ctx.config.get("bicep", "use_binary_from_path") == "false")
 
     @mock.patch("shutil.which")
     def test_run_bicep_command_raise_error_if_bicep_cli_not_found_when_use_binary_from_path_is_true(self, which_stub):
@@ -60,35 +88,6 @@ class TestBicep(unittest.TestCase):
         )
 
         cli_ctx.config.set_value("bicep", "use_binary_from_path", "false")
-
-    @mock.patch("os.chmod")
-    @mock.patch("os.stat")
-    @mock.patch("io.BufferedWriter")
-    @mock.patch("azure.cli.command_modules.resource._bicep.open")
-    @mock.patch("azure.cli.command_modules.resource._bicep.urlopen")
-    @mock.patch("os.path.exists")
-    @mock.patch("os.path.dirname")
-    @mock.patch("os.path.isfile")
-    def test_use_bicep_cli_from_path_false_after_install(self, isfile_stub, dirname_stub, exists_stub, urlopen_stub, open_stub, buffered_writer_stub, stat_stub, chmod_stub):
-        isfile_stub.return_value = False
-        dirname_stub.return_value = "tmp"
-        exists_stub.return_value = True
-        buffered_writer_stub.write.return_value = None
-
-        stat_result = mock.Mock()
-        stat_result.st_mode = 33206
-        stat_stub.return_value = stat_result
-
-        chmod_stub.return_value = None
-
-        response = mock.Mock()
-        response.getcode.return_value = 200
-        response.read.return_value = b"test"
-        urlopen_stub.return_value = response
-
-        ensure_bicep_installation(cli_ctx, release_tag="v0.14.85", stdout=False)
-
-        self.assertTrue(cli_ctx.config.get("bicep", "use_binary_from_path") == "false")
 
     @mock.patch("azure.cli.command_modules.resource._bicep._logger.warning")
     @mock.patch("azure.cli.command_modules.resource._bicep._run_command")

--- a/src/azure-cli/azure/cli/command_modules/resource/tests/latest/test_resource_bicep.py
+++ b/src/azure-cli/azure/cli/command_modules/resource/tests/latest/test_resource_bicep.py
@@ -28,7 +28,7 @@ class TestBicep(unittest.TestCase):
         isfile_stub.return_value = False
 
         with contextlib.suppress(FileNotFoundError):
-            remove_bicep_installation()
+            remove_bicep_installation(cli_ctx)
         with self.assertRaisesRegex(CLIError, 'Bicep CLI not found. Install it now by running "az bicep install".'):
             run_bicep_command(cli_ctx, ["--version"], auto_install=False)
 

